### ISSUE TITLE
docker-compose.yml entrypoint quotes misunderstood since VSCode 1.64.0

### ISSUE THREAD

**Issue by qdm12** [microsoft/vscode-remote-release#6337]
- VSCode Version: 1.64.2
- Local OS Version: Windows 10 Version 10.0.19042 Build 19042 (this also happened on Linux and OSX)
- Remote OS Version: Docker with any Debian or Alpine based images, using the `zsh` shell
- Remote Extension/Connection Type: Docker
- Logs:

```
Creating project_devcontainer_vscode_1 ... done
[5192 ms] Start: Run: docker ps -q -a --filter label=com.docker.compose.project=project_devcontainer --filter label=com.docker.compose.service=vscode
[5500 ms] Start: Run: docker inspect --type container c6cce158f477
[5792 ms] Start: Inspecting container
[5793 ms] Start: Run: docker inspect --type container c6cce158f4779e0bb4de5d14fa65a88364239ecee7d422688ff9b1419125d93b
[6102 ms] Start: Run in container: /bin/sh
[6116 ms] Start: Run in container: uname -m
[6423 ms] Start: Run in container: cat /etc/passwd
[6423 ms] Stdin closed!
[6424 ms] Shell server terminated (code: 1, signal: null)

Error response from daemon: Container c6cce158f4779e0bb4de5d14fa65a88364239ecee7d422688ff9b1419125d93b is not running
```

Steps to Reproduce:

I used to have a development container with

```json
    "dockerComposeFile": [
        "docker-compose.yml"
    ],
```

And in my docker-compose.yml I had the entrypoint

```yml
    entrypoint: zsh -c "while sleep 1000; do :; done"
```

This worked to keep the container running in order to have vscode exec into it, and used that for various Docker image with thousands of users.

Now the problem came with vscode 1.64.0 where the command run does not understood the whole command with double quotes. The container shows:

```sh
Container started
zsh:1: unmatched "
```

before exiting.

The fix was to have it as 

```yml
entrypoint: ["zsh", "-c", "while sleep 1000; do :; done"]
```

instead, although this breaks a lot of setups and compatibility.

Relevant issue where it was found: https://github.com/qdm12/latexdevcontainer/pull/25

To reproduce, you can use the following:

1. Download [devtainr](https://github.com/qdm12/devtainr/releases/tag/v0.6.0) as `devtainr`
2. Run `devtainr -dev base`
3. Change back the `entrypoint: ["zsh", "-c", "while sleep 1000; do :; done"]` to `entrypoint: zsh -c "while sleep 1000; do :; done"`
4. Open the VSCode command palette and select `Remote-Containers: Rebuilt Container`

**Comment by gabrielly-ramos** [comment]
I have same problem

**Comment by chrmarti** [comment]
This has slipped our attention because it only affect Docker Compose v1. A workaround is to use Docker Compose v2.

**Comment by qdm12** [comment]
It affects us on docker-compose v3. I have no idea about v1 or v2.

**Comment by chrmarti** [comment]
You are referring to the compose file version. The compose version depends on the CLI, you can check it with `docker-compose version`.

**Comment by qdm12** [comment]
Ah I understand, my apologies. Although I think a lot of people are still using docker-compose v1 as far as I know, shouldn't this be fixed for retrocompatibility? Or at least show a warning about it? It was quite troublesome to debunk the cause from the error log, since it looks like a Docker problem although it is caused by a VScode upgrade.

**Comment by chrmarti** [comment]
I have implemented a fix. Will let you know when it is published in a new version of the Remote-Containers extension.

**Comment by chrmarti** [comment]
Available with Remote-Containers 0.223.0 (pre-release).

**Comment by qdm12** [comment]
/verified

Awesome thanks @chrmarti for fixing this!
